examples/raycasting/main.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/png"
+	"log"
+	"math"
+
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+
+	"github.com/hajimehoshi/ebiten/inpututil"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/examples/resources/images"
+)
+
+const (
+	screenWidth  = 240
+	screenHeight = 240
+)
+
+var (
+	bgImage       *ebiten.Image
+	shadowImage   *ebiten.Image
+	triangleImage *ebiten.Image
+)
+
+func init() {
+	// Decode image from a byte slice instead of a file so that
+	// this example works in any working directory.
+	// If you want to use a file, there are some options:
+	// 1) Use os.Open and pass the file to the image decoder.
+	//    This is a very regular way, but doesn't work on browsers.
+	// 2) Use ebitenutil.OpenFile and pass the file to the image decoder.
+	//    This works even on browsers.
+	// 3) Use ebitenutil.NewImageFromFile to create an ebiten.Image directly from a file.
+	//    This also works on browsers.
+	img, _, err := image.Decode(bytes.NewReader(images.Tile_png))
+	if err != nil {
+		log.Fatal(err)
+	}
+	bgImage, _ = ebiten.NewImageFromImage(img, ebiten.FilterDefault)
+	shadowImage, _ = ebiten.NewImage(screenWidth, screenHeight, ebiten.FilterDefault)
+	triangleImage, _ = ebiten.NewImage(screenWidth, screenHeight, ebiten.FilterDefault)
+	triangleImage.Fill(color.White)
+}
+
+type Line struct {
+	X1, Y1, X2, Y2 float64
+}
+
+func directedRay(x, y, length, v float64) Line {
+
+	return Line{
+		X1: x,
+		Y1: y,
+		X2: x + length*math.Cos(v),
+		Y2: y + length*math.Sin(v),
+	}
+}
+
+// Algebra from wikipedia
```

Comment:
In Go convention, function comment should start with the function name like:

```go
// intersection calculations the intersection of given two lines.
```

Reply:
Ok! Added it. Is the wikipedia link ok? If so, should it be on the same line?

Reply:
Yes. or, you can put the Wikipedia link in the implementation. Both are fine.